app/hotels/src/singleHotel/SingleHotelContainer.js

```diff
@@ -81,32 +82,49 @@ type Props = {|
 export default class SingleHotelContainerWithContext extends React.Component<
   Props,
 > {
-  renderSingleHotelsContext = ({ currency }: HotelsState) => (
+  renderHotelsContext = ({ currency }: HotelsState) => (
     <SingleHotelContext.Consumer>
-      {this.renderHotelsContext(currency)}
+      {this.renderSingleHotelsContext(currency)}
     </SingleHotelContext.Consumer>
   );
 
-  renderHotelsContext = (currency: $PropertyType<HotelsState, 'currency'>) => ({
+  renderSingleHotelsContext = (
+    currency: $PropertyType<HotelsState, 'currency'>,
+  ) => ({
     checkin,
     checkout,
     roomsConfiguration,
     hotelId,
-  }: SingleHotelState) => (
-    <SingleHotelContainer
-      currency={currency}
-      hotelId={hotelId}
-      checkin={checkin}
-      checkout={checkout}
-      roomsConfiguration={roomsConfiguration}
-      {...this.props}
-    />
-  );
+    apiProvider,
+  }: SingleHotelState) => {
+    if (apiProvider === 'booking') {
+      return (
+        <SingleHotelContainer
```

Comment:
So we are passing the same props here, right? Just the wrapper component is changing. Would it be possible to have the logic do something like:
```
let Component = Stay22SingleHotel;
if (apiProvider === 'booking') {
  Component = SingleHotelContainer;
}
return (
  <Component {...props} />
);
```